Library/Homebrew/brew.sh

```diff
@@ -172,6 +172,11 @@ case "$@" in
     source "${HOMEBREW_LIBRARY}/Homebrew/list.sh"
     homebrew-list "$@" && exit 0
     ;;
+  # falls back to cmd/tap.rb on a non-zero return
+  tap)
+    source "${HOMEBREW_LIBRARY}/Homebrew/tap.sh"
+    homebrew-tap "$@" && exit 0
```

Comment:
How/when will this exit with a non-zero result? Not currently seeing any e.g. `return 1` in there?

Reply:
Sorry for not having been clear! What I meant is that the current implementation indeed never returns non-zero, and that's intended.

In previous iterations, there was argument handling logic in `homebrew-tap` that could return 1. However, as the function `homebrew-tap` is meant for zero-argument usage only, I removed that logic in e9b158cd5ca881cc3edd2c80f9cc537b11fcb1c4 and made sure at the call site that the function is only called when there are no `brew tap` arguments. I suppose your question is with the comment `falls back to cmd/tap.rb on a non-zero return` above, which I forgot to remove, so I removed it in ea011ae528bd1a67e3be56043b96c3da92073c75.

Reply:
> made sure at the call site that the function is only called when there are no `brew tap` arguments

Gotcha, I see that now. Might be a little clearer if this is above the wildcard commands and doesn't do `&& exit 0` but instead has `exit 0` be unconditional as there's no `return` status to check here.

Reply:
Done in c17154d2acebcfa085bd0df57d9e1902684c42c3.